src/replication/ng/model/execution.go

```diff
@@ -16,17 +16,38 @@ package model
 
 import "time"
 
+// ExecutionStatus is the status of an execution
+type ExecutionStatus string
+
+// TaskStatus is the status of a task
+type TaskStatus string
+
+// execution/task status const
+const (
+	ExecutionStatusFailed     ExecutionStatus = "Failed"
+	ExecutionStatusSucceed    ExecutionStatus = "Succeed"
+	ExecutionStatusStopped    ExecutionStatus = "Stopped"
+	ExecutionStatusInProgress ExecutionStatus = "InProgress"
+	TaskStatusFailed          TaskStatus      = "Failed"
+	TaskStatusSucceed         TaskStatus      = "Succeed"
+	TaskStatusStopped         TaskStatus      = "Stopped"
+	TaskStatusInProgress      TaskStatus      = "InProgress"
+	TaskStatusPending         TaskStatus      = "Pending"
+)
+
 // Execution defines an execution of the replication
 type Execution struct {
-	ID         int64     `json:"id"`
-	PolicyID   int64     `json:"policy_id"`
-	Total      int       `json:"total"`
-	Failed     int       `json:"failed"`
-	Succeed    int       `json:"succeed"`
-	Pending    int       `json:"pending"`
-	InProgress int       `json:"in_progress"`
-	StartTime  time.Time `json:"start_time"`
-	EndTime    time.Time `json:"end_time"`
+	ID         int64           `json:"id"`
+	PolicyID   int64           `json:"policy_id"`
+	Status     ExecutionStatus `json:"status"`
+	StatusText string          `json:"status_text"`
+	Total      int             `json:"total"`
+	Failed     int             `json:"failed"`
+	Succeed    int             `json:"succeed"`
+	Pending    int             `json:"pending"`
```

Comment:
here should match the status defined above. Failed Succeed Stoped InProgress.
And i see there is a column called 'trigger_mode' in UI design.  i think we should store it in execution data if this value is mutable in policy.